lib/honeybadger/client.ex

```diff
@@ -130,6 +130,9 @@ defmodule Honeybadger.Client do
 
       {:error, %Jason.EncodeError{message: message}} ->
         Logger.warn(fn -> "[Honeybadger] Notice encoding failed: #{message}" end)
+
+      {:error, %Protocol.UndefinedError{description: message}} ->
```

Comment:
This is failing dialyzer because the structure seems to be wrong.

```suggestion
      {:error, %Protocol.UndefinedError{message: message}} ->
```

Reply:
```
lib/honeybadger/client.ex:134: 
The pattern {'error', #{'description':=_message@2, '__struct__':='Elixir.Protocol.UndefinedError'}} 
can never match the type {
'error',#{'__exception__':=_, '__struct__':='Elixir.Jason.EncodeError', 'message':=binary()}} | {'ok',binary()}
```

The [Protocol.UndefinedError](https://github.com/elixir-lang/elixir/blob/5d57c9117494a97a6dd972d2334f19892aa0fafe/lib/elixir/lib/exception.ex#L1167) has description and not message.

I think the dialyzer error refers this [spec](https://github.com/honeybadger-io/honeybadger-elixir/blob/master/lib/honeybadger/json.ex#L4), which needs to be updated to return both errors